src/AffectList.hpp

```diff
@@ -1,48 +1,36 @@
 #pragma once
 
 #include "AFFECT_DATA.hpp"
-#include "GenericListIter.hpp"
-
-#include <utility>
+#include "GenericList.hpp"
 
 class AffectList {
-    AFFECT_DATA *first_{};
+    GenericList<AFFECT_DATA> list_;
```

Comment:
Now that we have GenericList,  do we need to have AffectList wrapper type any more? i.e. could we be using `GenericList<AFFECT_DATA>` everywhere?

Reply:
I wondered about this. There's a couple of useful helper functions on the AffectList that were "nice" to keep for now, mainly "find by skill". But I agree I think this could probably go. In fact, looking at `find_by_skill` : 

```cpp
    if (auto it = ranges::find(*this, skill_number, [](auto &af) { return af.type; }); it != end())
        return &*it;
    return nullptr;
```

this is an example of the code before I knew I could shortcut the lambda to "project" an affect to its skill. So now I could write this as:

```cpp
    if (auto it = ranges::find(*this, skill_number, &AFFECT_DATA::type); it != end())
        return &*it;
    return nullptr;
```

...at which point the function is "simple enough" to maybe inline where it's used. Maybe? Sometimes it's nice to have a human-friendly name for these operations. We can make free functions that take `GenericList<AFFECT_DATA>` and find by skill I guess, so callers would do:

```cpp
auto *af = find_by_skill(list_, skill_number);
```
in that case. More "FP" than OO :)

Either way; for now I'll leave this I think and we can always remove it later (which I'm pretty sure we should do on balance).